torch/nn/parallel/distributed.py

```diff
@@ -1234,8 +1234,8 @@ def _assign_modules_buffers(self):
         }
 
     def _build_debug_param_to_name_mapping(self, parameters):
-        if dist.get_debug_level() == dist.DebugLevel.OFF:
-            return {}
+        # if dist.get_debug_level() == dist.DebugLevel.OFF:
+        #     return {}
```

Comment:
@rohan-varma, is it okay to remove this check? Param to name mapping is needed for sparse allreduce when global unique id is passed in. I assume cost to create this mapping is negligible?

Reply:
yep